src/v/cloud_storage/topic_manifest.cc

```diff
@@ -41,68 +213,34 @@ ss::future<> topic_manifest::update(ss::input_stream<char> is) {
     co_await ss::copy(is, os);
     iobuf_istreambuf ibuf(result);
     std::istream stream(&ibuf);
-    json::Document m;
+
     rapidjson::IStreamWrapper wrapper(stream);
-    m.ParseStream(wrapper);
-    update(m);
-    co_return;
-}
+    rapidjson::Reader reader;
+    topic_manifest_handler handler;
+    if (reader.Parse(wrapper, handler)) {
```

Comment:
Is it worth catching some of the throws here (e.g., from invalid compaction strategy)? It should allow error messages to be improved.

Reply:
moved `lexical_cast` here